apps/core/models.py

```diff
@@ -0,0 +1,82 @@
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from dive.base_models import BaseModel, NamedModelMixin
+
+
+class Dataset(BaseModel, NamedModelMixin):
+    class DatasetStatus(models.TextChoices):
+        PENDING = "pending", _("Pending")
+        UPLOADED = "uploaded", _("Uploaded")
+        EXTRACTED = "extracted", _("Extracted")
+
+    file_url = models.TextField()
```

Comment:
Lets make this ` CharField`

Reply:
The urls can be pretty long and don't generally have max length limitations(although specs say it's around 2000 chars). So instead of having such a long max_length I think we can stick with Text field. Also, Postgres has no performance differences between the two fields.